electronicparsers/openmx/parser.py

```diff
@@ -767,6 +805,14 @@ def parse(self, mainfile: str, archive: EntryArchive, logger):
                             scf.energy = Energy(
                                 sum_eigenvalues=EnergyEntry(value=u_ele * units.hartree)
                             )
+                    mainfile_uele = scf_steps[0].get('Uele')
+                    if uele[i] == mainfile_uele:
```

Comment:
Is there indeed the same precision printed in both mainfile (I'm not sure right now if Uele actually comes from mainfile or from the MD file...) and stdout? We are comparing floats here, so I think we should probably use something like `abs(Uele1 - Uele2) < desired_precision` to be safe, where desired precision would be based on the number of places for the less precise one.

Reply:
Also if we fail the check I would add a logger warning that the stdout is inconsistent with the calculation

Reply:
Also, `uele[i]` can be undefined at this point, right?

Reply:
> Is there indeed the same precision printed in both mainfile (I'm not sure right now if Uele actually comes from mainfile or from the MD file...) and stdout? We are comparing floats here, so I think we should probably use something like `abs(Uele1 - Uele2) < desired_precision` to be safe, where desired precision would be based on the number of places for the less precise one.

For this purpose, I added two functions - number_of_decimal_places() to get the precision and is_close() to compare the floats. Is it what you meant?

Reply:
Sorry for not being more precise, I could have saved you some work. I was in fact suggesting to just look into stdout and mainout file, look how many places we have and hardcode a reasonable value based on that, doing it at runtime is an overkill.